src/workers/cdc/primary.ts

```diff
@@ -740,6 +744,48 @@ const onPostRelationChange = async (
   }
 };
 
+const onMarketingCtaChange = async (
+  con: DataSource,
+  data: ChangeMessage<MarketingCta>,
+) => {
+  if (data.payload.op !== 'u') {
+    return;
+  }
+
+  const users = await con.getRepository(UserMarketingCta).findBy({
+    marketingCtaId: data.payload.after.campaignId,
+    readAt: IsNull(),
+  });
+
+  if (users.length > 0) {
+    await deleteRedisKey(
+      ...users.map((user) =>
+        generateStorageKey(
+          StorageTopic.Boot,
+          StorageKey.MarketingCta,
+          user.userId,
```

Comment:
Can't we get away with doing a single up? (so avoid the userId to do it for everyone)

Reply:
The cache key is `boot:marketing_cta:XDCZD-PHG`, so if we delete with the `boot:marketing_cta` prefix, we would delete keys for users who might now be in the campaign.

Reply:
Hmm that might be a interesting nuance, eg what happens if you in 2 campaigns?

Reply:
You only see one campaign at a time

Reply:
Yes see one, but you could technically be spooled for one + another one right?

Reply:
Yeah. You'd just re-fill the redis cache on your next boot request. It would still be the first in the queue